### PR TITLE
Updating criterion.rs to work with rust nightly 2015-11-27

### PR DESCRIPTION
**Pull request by faern**
In response to issue #55. I'm not sure every fix does exactly what it's supposed to. I mostly did my best to make it build.

This PR changes some of the dependencies to other forks of the dependencies that are more updated. These changes should of course be merged to the upstream dependencies and this crate be pointed back to them, but I started here and we can work on it from within this PR maybe?

### CODE REVIEW

File: src/routine.rs [hunk 1]
```diff
@@ -24,17 +22,20 @@ pub trait Routine {
...
         // Solve: [d + 2*d + 3*d + ... + n*d] * met = m_ns
-        let d = (2. * m_ns as f64 / met / (n * (n + 1)) as f64).ceil() as u64;
+        let total_runs = n * (n + 1) / 2;
+        let d = (m_ns as f64 / met / total_runs as f64).ceil() as u64;
 
-        let m_iters = iter::iterate(d, |a| a + d).take(n).collect::<Vec<u64>>();
+        let m_iters = (1..n as u64).map(|a| a*d).collect::<Vec<u64>>();
 
-        let m_ns = m_iters.iter().map(|&x| x).sum::<u64>() as f64 * met;
+        let m_ns = total_runs as f64 * met;
```
Comment (japaric) [inline src/routine.rs]: This should be `let m_ns = total_runs as f64 * d as f64 * met`

**Comment by japaric** [comment]
I've looked over these changes and the forked dependencies, and all the updates seem to preserve the behavior except for the one that I pointed out here. I'd be in favor of merging this PR as a first step forward if @laumann is okay with us using his commits.

**Comment by laumann** [comment]
@japaric @faern No problem with me :smile: Glad that you can use it...

**Comment by japaric** [comment]
@laumann Great, thanks :smile:

**Pull request merged**